easybuild/easyblocks/t/tbb.py

```diff
@@ -224,21 +246,26 @@ def sanity_check_step(self):
                 os.path.join('lib', 'cmake', 'TBB', 'TBBConfigVersion.cmake'),
             ])
 
-        super(EB_tbb, self).sanity_check_step(custom_paths=custom_paths)
+        if self.cfg['with_python']:
+            custom_paths['dirs'].append(os.path.join(self.tbb_subdir, 'python'))
+            custom_commands.extend(['python -c "import tbb"'])
+
+        super(EB_tbb, self).sanity_check_step(custom_paths=custom_paths, custom_commands=custom_commands)
 
     def make_module_extra(self):
         """Add correct path to lib to LD_LIBRARY_PATH. and intel license file"""
         txt = super(EB_tbb, self).make_module_extra()
 
         if self.toolchain.is_system_toolchain():
-            tbb_subdir = 'tbb'
-            txt += self.module_generator.prepend_paths('CPATH', [os.path.join(tbb_subdir, 'include')])
-        else:
-            tbb_subdir = ''
+            txt += self.module_generator.prepend_paths('CPATH', [os.path.join(self.tbb_subdir, 'include')])
+
+        root_dir = os.path.join(self.installdir, self.tbb_subdir)
+        txt += self.module_generator.set_environment('TBBROOT', root_dir)
+        # TBB_ROOT used e.g. by FindTBB.cmake
+        txt += self.module_generator.set_environment('TBB_ROOT', root_dir)
 
-        txt += self.module_generator.set_environment('TBBROOT', os.path.join(self.installdir, tbb_subdir))
-        # Used e.g. by FindTBB.cmake
-        txt += self.module_generator.set_environment('TBB_ROOT', os.path.join(self.installdir, tbb_subdir))
+        if self.cfg['with_python']:
+            txt += self.module_generator.prepend_paths('PYTHONPATH', [os.path.join(root_dir, 'python')])
```

Comment:
`prepend_paths` doesn't like being fed absolute paths (although you can force it), so this should be changed to:

```suggestion
            txt += self.module_generator.prepend_paths('PYTHONPATH', [os.path.join(self.tbb_subdir, 'python')])
```